common/transport/DcgmIpc.cpp

```diff
@@ -579,8 +595,21 @@ void DcgmIpc::ConnectTcpAsyncImpl(DcgmIpcConnectTcp &tcpConnect)
     bufferevent_setcb(bev, DcgmIpc::StaticReadCB, NULL, DcgmIpc::StaticEventCB, this);
     bufferevent_enable(bev, EV_READ | EV_WRITE);
 
+    /* Allow IPv6 IPs to override family */
+    char buf[16];
+    sa_family_t family = AF_INET;
+    std::string str = tcpConnect.m_hostname;
+    if (str.size() > 1 && str[0] == '[' && str[str.size() - 1] == ']')
```

Comment:
Could there be a possible out-of-bounds read/copy here?
For str.size() - 2 to make sense in the assignment to m_hostname below, I think str.size() needs to be at least 3.
```suggestion
    if (str.size() >= 3 && str[0] == '[' && str[str.size() - 1] == ']')
```